File: ios/recargaPay/DeviceLanguageModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(DeviceLanguageModule, NSObject)
RCT_EXTERN_METHOD(getCurrentLanguage:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
@end

File: ios/recargaPay/DeviceLanguageModule.swift
import Foundation
import React

@objc(DeviceLanguageModule)
class DeviceLanguageModule: NSObject {
  @objc(getCurrentLanguage:rejecter:)
  func getCurrentLanguage(
    resolve: RCTPromiseResolveBlock,
    rejecter reject: RCTPromiseRejectBlock
  ) {
    let language = Locale.preferredLanguages.first ?? "unknown"
    resolve(language)
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    false
  }
}
